### ISSUE TITLE
Checklists and Merged IDs

### ISSUE THREAD

**Issue by ThunderSon** [OWASP/wstg#640]
Should we keep the IDs as normal for tests that are now merged or removed?

Maybe we can set a placeholder instead, or have a special value that we can parse for later terms.

`WSTG-INFO-09` is set twice, I am guessing there's a typo, just making sure it wasn't set specifically.

**Comment by kingthorin** [comment]
I think that must be a parsing issue due to: https://github.com/OWASP/wstg/blob/master/document/4-Web_Application_Security_Testing/01-Information_Gathering/09-Fingerprint_Web_Application.md

Uh, where are you seeing INFO-09 twice?
https://github.com/OWASP/wstg/blob/master/checklist/checklist.json seems fine...

**Comment by kingthorin** [comment]
It does seem to be a typo. The placeholders should be left for the time being (IMHO).

![image](https://user-images.githubusercontent.com/7570458/100395197-05ce1780-300e-11eb-9643-7a768c39abd7.png)

(The first of each set should be INFO-08.)

**Comment by ThunderSon** [comment]
There's the Buffer Overflow one as well, INPV-13. That solution that sounds fair, changing the description, but that won't be replicated in the JSON sadly. That's what I meant by setting a certain "value" that we can parse based on it.
If "value" is set, grab the referenced test in the md file and set the known description for merged tests.

**Comment by kingthorin** [comment]
Ya if there's some sort of element we want to set on those placeholder pages I'm good with that.

**Comment by ThunderSon** [comment]
@rejahrehim What do you feel about this?
This is not as well a pre-requirement for v4.2, so no big rush. We'll just update the typos in the checklists for now

**Comment by rejahrehim** [comment]
Yes. We could keep a common heading which I could parse. If So, we can update the checklist with it. 
Also, We should keep a specific format in the checklist to identify expired test cases. Like ` Expired since v4.2`

**Comment by himasree9** [comment]
Hi, I reviewed the current state of the repository regarding this issue.
WSTG-INFO-09 is no longer duplicated in checklist.json, so that part seems resolved.
However, the markdown file indicates that it has been merged into WSTG-INFO-08, while it still appears as an active test in the JSON without any “retired” or “merged” indication.

Additionally, I noticed that WSTG-INPV-13 appears twice in checklist.json (Buffer Overflow and Format String Injection), which may indicate another structural inconsistency.

Should we standardize how merged or retired tests are represented in the checklist (e.g., adding a specific flag or format)? I’d be happy to help if this is still something we want to address.

**Comment by kingthorin** [comment]
@HimasreeKolathur24 It's something I haven't looked at in ages. It's related to: https://github.com/OWASP/wstg/issues/1165

If you feel like digging into it and presenting a plan here I'm good with that. You'll need to check content or PR history to see what's been merged/dropped. I think there was another over the weekend.

You'll also need to look at the checklist generation workflow/scripts and see if we purposefully worked around one/some and not others....

I'll assign you for now, but if we can't nail down a way forward or it becomes too much work I can unassign later, just let me know.

Please do plan/discuss before diving into the changes/work though.

**Comment by himasree9** [comment]
@kingthorin,
After reviewing the repository:

• I found 4 markdown files that explicitly state “This content has been merged into…”.
• These tests still exist as active entries in checklist.json.
• There is currently no structured indicator (e.g., retired, merged_into, or similar field) to distinguish placeholders from active tests.
• ID continuity appears to be preserved intentionally, but lifecycle state is implicit rather than explicit.

Proposed direction:
Instead of removing placeholders, we could introduce a consistent structured representation in checklist.json, such as:

{
  "id": "WSTG-INFO-09",
  "name": "Fingerprint Web Application",
  "status": "merged",
  "merged_into": "WSTG-INFO-08"
}

This would:
• Preserve ID continuity
• Allow parsing automation to distinguish active vs merged tests
• Keep checklist generation predictable
• Avoid relying on manual markdown phrasing

Happy to refine this approach based on workflow constraints.

**Comment by kingthorin** [comment]
>  Avoid relying on manual markdown phrasing

How? The checklists are built from the markdown.

**Comment by himasree9** [comment]
Thanks for the clarification.Since the checklist is generated from markdown, it would be better to standardize lifecycle information there rather than rely on the current phrasing.
Would introducing a small, consistent metadata indicator for merged tests (instead of a free-text sentence) make parsing more explicit while keeping markdown as the source of truth?

**Comment by kingthorin** [comment]
We still need to be more specific. We kind of have a very limited options.

- We could use an HTML comment (though I hate HTML in markdown).
- We could use an un-used reference link. Ex: `[merged]: # (WSTG-ERRH-02)` We can leave the actual sentence/markdown content, then have this as the list thing in the file (or well second, as we should end with a blank line). Expect `merged` as the keyword and the destination/replacement scenario ID as the (comment).

I think I prefer the second one. We can also document it in the template info somewhere. Maybe just https://github.com/OWASP/wstg/tree/master/template/README.md

I'm open to other ideas...

[comment]: # (This is a comment that will not be rendered.)

**Comment by kingthorin** [comment]
PoC:

<img width="513" height="515" alt="Image" src="https://github.com/user-attachments/assets/54506da9-682b-4a86-b3f3-a7d09c125c94" />

**Comment by himasree9** [comment]
Thanks for the PoC — the unused reference link approach looks clean and keeps everything valid markdown.
I agree it’s preferable to HTML comments and gives us something explicit and parseable.
If we standardize something like:

[merged]: # (WSTG-INFO-08)

at the end of merged files and document it in the template README, that should give us a consistent pattern.
If that sounds good to you, I can proceed with identifying all current merged tests and normalizing them accordingly.

**Comment by kingthorin** [comment]
Sure go for it!

Thanks for talking through it :grinning:
